lib/io/can.hpp

```diff
@@ -41,13 +41,44 @@ struct sockaddr_can {
 
 class ICanProcessor {
  public:
-  virtual void processMessage(const io::CanFrame &frame) = 0;
+  /**
+   * @brief Called by ICan::receive() when a message with the given ID is received
+   * @param frame The CAN message to be processed
+   * @return core::Result::kSuccess if the message was processed successfully,
+   * core::Result::kFailure otherwise
+   */
+  virtual core::Result processMessage(const io::CanFrame &frame) = 0;
 };
 
 class ICan {
  public:
-  virtual core::Result send(const CanFrame &message)                                          = 0;
-  virtual core::Result receive()                                                              = 0;
+  /**
+   * @brief Send a CAN message
+   * @param message The message to be sent
+   * @return core::Result::kSuccess if the message was sent successfully core::Result::kFailure
+   * otherwise
+   * @note To send an extended CAN message, set the EFF flag in the can_id field by ORing it with
+   * CAN_EFF_FLAG
+   */
+  virtual core::Result send(const CanFrame &message) = 0;
+
+  /**
+   * @brief Attempt to receive a CAN message
+   * @return core::Result::kSuccess if a message was received and successfully processed by all
+   * registered processors, core::Result::kFailure if a message is not received successfully or if a
+   * message is received but not processed by all registered processors
+   * @note If a message is received then the message will be processed by all processors registered
+   * to that ID
+   * @note If the message is an extended CAN message, the EFF flag will be set in the can_id field
+   * and thus the actual ID will be frame.can_id - 0x80000000
+   */
+  virtual core::Result receive() = 0;
+
+  /**
+   * @brief Registers a processor to be called when a message with the given ID is received
+   * @param id The ID of the message to be processed
+   * @param processor The processor to be called when a message with the given ID is received
+   */
   virtual void addProcessor(const std::uint16_t id, std::shared_ptr<ICanProcessor> processor) = 0;
```

Comment:
Can this not fail too? Not 100% sure on context here but if this could fail then a `core::Result` return value might make sense here too?